specification/cognitiveservices/OpenAI.Inference/models/completions/azure_content_filtering.tsp

```diff
@@ -0,0 +1,175 @@
+import "@typespec/http";
+import "@typespec/rest";
+import "@typespec/versioning";
+import "@azure-tools/typespec-azure-core";
+
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using TypeSpec.Versioning;
+
+namespace Azure.OpenAI;
+
+@added(ServiceApiVersions.v2023_06_01_Preview)
+@doc("Ratings for the intensity and risk level of harmful content.")
+enum ContentFilterSeverity {
+  @doc("""
+  Content may be related to violence, self-harm, sexual, or hate categories but the terms 
+  are used in general, journalistic, scientific, medical, and similar professional contexts, 
+  which are appropriate for most audiences.
+  """)
+  safe: "safe",
+
+  @doc("""
+  Content that expresses prejudiced, judgmental, or opinionated views, includes offensive 
+  use of language, stereotyping, use cases exploring a fictional world (for example, gaming,
+  literature) and depictions at low intensity.
+  """)
+  low: "low",
+
+  @doc("""
+  Content that uses offensive, insulting, mocking, intimidating, or demeaning language 
+  towards specific identity groups, includes depictions of seeking and executing harmful 
+  instructions, fantasies, glorification, promotion of harm at medium intensity.
+  """)
+  medium: "medium",
+
+  @doc("""
+  Content that displays explicit and severe harmful instructions, actions, 
+  damage, or abuse; includes endorsement, glorification, or promotion of severe 
+  harmful acts, extreme or illegal forms of harm, radicalization, or non-consensual 
+  power exchange or abuse.
+  """)
+  high: "high",
+}
+
+@added(ServiceApiVersions.v2023_06_01_Preview)
+@doc("Information about filtered content severity level and if it has been filtered or not.")
+model ContentFilterResult {
+  @doc("Ratings for the intensity and risk level of filtered content.")
+  @projectedName("json", "severity")
+  severity: ContentFilterSeverity;
+
+  @doc("A value indicating whether or not the content has been filtered.")
+  @projectedName("json", "filtered")
+  filtered: boolean;
```

Comment:
Do you think we may want to use `contentFilterResultBase` (notice singular `result`) verbatim from what there is in the service? It would be a model that looks like:

```ts
model ContentFilterResultBase {
  filtered: boolean;
}
```

Then wherever (well, not wherever, but I will mark the places where the swagger seems to use a spread operator) we have `filtered` we use instead `...ContentFilterResultBase`